src/main/java/com/powsybl/openloadflow/network/AbstractLfBranch.java

```diff
@@ -39,6 +46,14 @@ protected AbstractLfBranch(LfBus bus1, LfBus bus2, PiModel piModel) {
         this.piModel = Objects.requireNonNull(piModel);
     }
 
+    protected static SortedSet<CurrentLimits.TemporaryLimit> getSortedTemporaryLimits(CurrentLimits currentLimits) {
```

Comment:
Indeed you have change the initial code too much for me. The values of current limits attached to a LfBranch sould be stored in per unit system. This `<CurrentLimits.TemporaryLimit>` should disappear too I think.

Reply:
Why all these treatments have disappear: 
`temporaryLimit.getValue() != Double.NaN ? temporaryLimit.getValue() * getBus1().getNominalV() / PerUnit.SB : Double.NaN);`

Reply:
Alright. Then we could introduce a new `LfTemporaryLimit` class. Or come back to the map.

Reply:
(and the treatments you're referring to have disappeared as it's the ones translating in per unit)